projects/WTI-UI/src/app/modules/shared/components/problem-selector/problem-selector.component.ts

```diff
@@ -2,97 +2,162 @@ import { Component, forwardRef, OnInit, OnDestroy, Input } from '@angular/core';
+	//PC2 contest clock.  This causes the Component to execute its "loadProblems()" method; that method in turn checks
+	//whether the contest is currently RUNNING; if so, it loads the problem names; if not, it blanks out problem names.
 
-  ngOnDestroy(): void {
-    this._unsubscribe.next();
-    this._unsubscribe.complete();
-  }
+		this._contestService.contestClockEvent
+		.pipe(takeUntil(this._unsubscribe))
+		.subscribe(_ => this.loadProblems());
+	}
 
-  registerOnChange(fn: (event: any) => void) {
-    this.onChange = fn;
-  }
+	ngOnDestroy(): void {
+		this._unsubscribe.next();
+		this._unsubscribe.complete();
+	}
 
-  registerOnTouched(fn: (event: any) => void) {
-    this.onTouched = fn;
-  }
+	registerOnChange(fn: (event: any) => void) {
+		this.onChange = fn;
+	}
 
-  writeValue(value: string) {
-    this.value = (value === null) ? undefined : value;
-  }
+	registerOnTouched(fn: (event: any) => void) {
+		this.onTouched = fn;
+	}
 
-  private loadProblems(): void {
-	  if (DEBUG_MODE) {
-		  console.log ("Executing ProblemSelectorComponent.loadProblems()") ;
-	  }
-    if (this._contestService.isContestRunning) {
-    	if (DEBUG_MODE) {
-    		console.log ("ProblemSelectorComponent.loadProblems(): ContestService.isContestRunning() returned positive Truthy value") ;
-    	}
-      this._contestService.getProblems()
-      .pipe(takeUntil(this._unsubscribe))
-      .subscribe((data: ContestProblem[]) => {
-    	  if (DEBUG_MODE) {
-    		  console.log ("ProblemSelectorComponent.loadProblems(): subscription callback from ContestService.getProblems() returned data:");
-    		  console.log (data) ;
-    	  }
-        this.problems = data;
-      }, (error: any) => {
-    	  if (DEBUG_MODE) {
-    		  console.log ("ProblemSelectorComponent.loadProblems(): subscription callback from ContestService.getProblems() returned error");
-    		  console.log ("  Setting contest problem list to empty array." ) ;
-    	  }
-        this.problems = [];
-      });
-    } else {
-    	if (DEBUG_MODE) {
-    		console.log ("ProblemSelectorComponent.loadProblems(): ContestService.isContestRunning() returned negative Truthy value") ;
-			console.log ("  Setting contest problem list to empty array." ) ;
+	writeValue(value: string) {
+		this.value = (value === null) ? undefined : value;
+	}
 
-    	}
-      this.problems = [];
-    }
-  }
+	/**
+	 * This method determines whether the contest problems should be displayed in a "ProblemSelectorComponent".
+	 * Problems are only allowed to be displayed when the contest has been started, which is determined by looking
+	 * at the "wallClockStartTime" value in the current ContestClock.  The method fetches the current clock from the WTI-API server,
+	 * and if wallClockStartTime is greater than zero it fetches the current problem list from the server and displays it in the
+	 * component; otherwise it sets the list of displayed problems to an empty list (array).
+	 */
+	private loadProblems(): void {
```

Comment:
woah. Is it possible to simplify this method through having multiple methods breaks etc.

Reply:
As with the previous comment, I don't see any way to use `break`s to address this.  As before, I think the problem was the large number of "debug" `if`s in the code, making it hard to see the structure.  I've eliminated all those debug statements; hopefully that makes the code more readable.  I'll welcome any suggestion for how to use `break`s to simplify it further.

Reply:
I think the in-line function i what clutters it up.  Folks love doing that in javascript too, and it makes it very hard to follow and read.  I guess it's the nature of the beast.

Reply:
I meant return sorry. Maybe the method can be terminated early if it is not going to lead anywhere?

Reply:
As noted, I've removed the large number of "debug" statements; hopefully this makes the code clearer.  Also, as @johnbrvc noted, the use of JavaScript "lambda functions" makes it a little hard to read.  (I didn't write that code originally, and I'm reluctant to change it as part of this PR.  If you think it still needs to be changed, I suggest submitting a separate Issue for cleaning that up...)